src/utils.c

```diff
@@ -126,7 +127,7 @@ ts_time_value_to_internal(Datum time_val, Oid type_oid)
 		if (ts_type_is_int8_binary_compatible(type_oid))
 			return DatumGetInt64(time_val);
 
-		elog(ERROR, "unknown time type OID %d", type_oid);
+		elog(ERROR, "unknown time type \"%s\"", format_type_be(type_oid));
```

Comment:
I think there are two cases here:

1. The type exists but is not a valid time type --- all is good and this error makes sense
2. The type does not exist, and then `format_type_be` will throw another error.

In particular for the second case, is this what we want here?

I think this concern applies to a lot of other suggested changes in this PR.

Reply:
hmm i could have sworn format_type_be returns OID when it cannot resolve but it does not and while there is a flag to make it not fail when it cannot resolve it returns '???' instead of the OID in that case. I'll revert the format_type_be changes.

Reply:
Actually regarding your 2nd point i dont think it applies this is all internal code and not being called with random type ids but with actual oids handed to us from postgres so the types are validated but its just that its a type we do not support.